Track payment cancelation and rename payment flow start event

kidinov opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #6786
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
* Renames `CARD_PRESENT_COLLECT_PAYMENT_TAPPED` -> `PAYMENTS_FLOW_ORDER_COLLECT_PAYMENT_TAPPED`
* Renames `SIMPLE_PAYMENTS_FLOW_CANCELED` -> `PAYMENTS_FLOW_CANCELED`
* Adds `PAYMENTS_FLOW_CANCELED` tracking when
    * Selection payment method fragment is closed during order payment flow (arg `flow=order_payment`)
    * Back click on the main simple payment fragment (arg `flow=simple_payment`)

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
* Use logcat with filtering by `Tracked`
* Go via Simple payments flow and Order payment flow and notice tracked events

kidinov committed:
Merge branch 'feature/ducks-in-row' into 6786-track-payment-cancelation-and-rename-payment-flow-start-event

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreationRepository.kt

kidinov committed:
Reanmed CARD_PRESENT_COLLECT_PAYMENT_TAPPED to PAYMENTS_FLOW_ORDER_COLLECT_PAYMENT_TAPPED

kidinov committed:
Renamed SIMPLE_PAYMENTS_FLOW_CANCELED to PAYMENTS_FLOW_CANCELED

kidinov committed:
Added cancelation event tracking where needed with tests

kidinov committed:
Merge branch 'feature/ducks-in-row' into 6786-track-payment-cancelation-and-rename-payment-flow-start-event

AnirudhBhat commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/simplepayments/SimplePaymentsFragmentViewModel.kt:
```diff
@@ -183,6 +183,10 @@ class SimplePaymentsFragmentViewModel @Inject constructor(
     }
 
     fun onBackButtonClicked() {
+        AnalyticsTracker.track(
```
❓ What do you think about writing tests to verify this in `SimplePaymentsFragmentViewModelTests`

kidinov replied:
In order to test it I need to refactor that which probably is not the goal of the PR?

AnirudhBhat replied:
Ahh! Didn't notice it was `AnalyticsTracker`. Completely fine! Let's not do it in this PR 👍

AnirudhBhat commented:
Thank you for working on this @kidinov! I've tested the changes and it works as expected. 

I noticed a few things:
1. `PAYMENTS_FLOW_ORDER_COLLECT_PAYMENT_TAPPED` event's "country" property will be "Unknown" if fired before IPP onboarding. I believe this happens even in the production as well. 
2. When the card reader is being scanned for devices, pressing the back button will not trigger `PAYMENTS_FLOW_CANCELED` event. However, clicking on the "Cancel" button on the same dialog triggers this event. Is this expected?
3. When Simple payments flow is started we get this event - `simple_payments_flow_started`. Once we click on the "Done" button and navigate to the "TakePayment" screen, we again fire `payments_flow_canceled`. Is this expected?

kidinov committed:
Do not track extra properties for `PAYMENTS_FLOW_ORDER_COLLECT_PAYMENT_TAPPED` event

kidinov committed:
Fixed wrongly tracked cancelation event on every dialog dismiss

kidinov commented:
@AnirudhBhat thank you for the review and valuable feedback 🙏

> PAYMENTS_FLOW_ORDER_COLLECT_PAYMENT_TAPPED event's "country" property will be "Unknown" if fired before IPP onboarding. I believe this happens even in the production as well.

Yep, Josh also pointed to this! I removed all extra "card reader" related fields from this event [37987a4](https://github.com/woocommerce/woocommerce-android/pull/6788/commits/37987a4f096e9b4306c39d3d3cf358c504d70f30)

> When the card reader is being scanned for devices, pressing the back button will not trigger PAYMENTS_FLOW_CANCELED event. However, clicking on the "Cancel" button on the same dialog triggers this event. Is this expected?

Is this about `payments_flow_failed` event?  I think this is a bug that was there for quite some time. Created [an issue](https://github.com/woocommerce/woocommerce-android/issues/6794)

> When Simple payments flow is started we get this event - simple_payments_flow_started. Once we click on the "Done" button and navigate to the "TakePayment" screen, we again fire payments_flow_canceled. Is this expected?

That's me being stupid 🤦‍♂️ . Thank you for catching this! Fixed here - 628885124b93b869ef12ca17a61cb33859d24120

AnirudhBhat reviewed (approved):
Thank you for all the changes! LGTM! :shipit:
